source/Calamari.Common/Features/Packages/ExtractPackage.cs

```diff
@@ -23,18 +23,25 @@ public ExtractPackage(ICombinedPackageExtractor combinedPackageExtractor, ICalam
             this.log = log;
         }
 
+        public void ExtractToCustomDirectory(PathToPackage? pathToPackage, string directory)
+        {
+            ExtractToCustomSubDirectory(pathToPackage, directory, null, null);
+        }
+
         public void ExtractToStagingDirectory(PathToPackage? pathToPackage, IPackageExtractor? customPackageExtractor = null)
         {
-            var targetPath = Path.Combine(Environment.CurrentDirectory, "staging");
-            fileSystem.EnsureDirectoryExists(targetPath);
-            Extract(pathToPackage, targetPath, PackageVariables.Output.InstallationDirectoryPath, customPackageExtractor);
+            ExtractToCustomSubDirectory(pathToPackage,
+                Path.Combine(Environment.CurrentDirectory, "staging"),
+                customPackageExtractor, null);
         }
 
         public void ExtractToStagingDirectory(PathToPackage? pathToPackage, string extractedToPathOutputVariableName)
         {
-            var targetPath = Path.Combine(Environment.CurrentDirectory, "staging");
-            fileSystem.EnsureDirectoryExists(targetPath);
-            Extract(pathToPackage, targetPath, extractedToPathOutputVariableName, null);
+            ExtractToCustomSubDirectory(
+                pathToPackage,
+                Path.Combine(Environment.CurrentDirectory, "staging"),
```

Comment:
Maybe we can promote `"staging"` and `"grouped"` to SpecialVariables and use that instead of string literals? I'm imagining it would be easier to understand later because we can do "find usages" on these variables. Also you can reuse these from tests.

Reply:
Yep good idea 👍

Reply:
I'm not sure if SpecialVariables is the right place because that's referring to Variable names for `IVariables`. I've moved the literals to constants but just at the top of the relevant class for now. I think if 'grouped' is used elsewhere later we can promote it further but at the moment it's only used in this class. Staging is used in `ExtractPackage` and `KubernetesDeploymentCommandBase` (and maybe some other places but I'm not going to update them).